src/Resource/Metadata/Application.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LAG\AdminBundle\Resource\Metadata;
+
+class Application
+{
+    public function __construct(
+        private string $name,
+        private ?string $dateFormat = null,
+        private ?string $timeFormat = null,
+        private ?string $translationDomain = null,
+        private ?string $translationPattern = null,
+        private ?string $routePattern = null,
+    ) {
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function withName(string $name): self
+    {
+        $self = clone $this;
+        $self->name = $name;
+
+        return $self;
+    }
+
+    public function getDateFormat(): ?string
+    {
+        return $this->dateFormat;
+    }
+
+    public function withDateFormat(string $dateFormat): self
+    {
+        $self = clone $this;
+        $self->dateFormat = $dateFormat;
+
+        return $self;
+    }
+
+    public function getTimeFormat(): ?string
+    {
+        return $this->timeFormat;
+    }
+
+    public function withTimeFormat(string $timeFormat): self
+    {
+        $self = clone $this;
+        $self->timeFormat = $timeFormat;
+
+        return $self;
+    }
+
+    public function getTranslationDomain(): ?string
```

Comment:
Coulld `Translation` be a value object with a domain and a pattern property ?

Reply:
It could, but it will require that the end-user to instanciate another object instead of just a string.

Reply:
Documentation has been added

Reply:
Who is the end-user ? If it's a developer just a craft add a factory to do the heavy lifting. 
"Instancianting" another object shouldn't be an issue, and it wasn't just one string but many, also it might have good side effects when you'll use the translation somewhere !

Reply:
Sorry, I mean the developer. It is better to not require a factory here as every metadata object can be used as php Attributes, and should be instanciable without services.

Reply:
For now, I think we keep as is, but it can be put in a future version